recipes-ni/ni-test-boot-time/files/upload_results.py

```diff
@@ -0,0 +1,137 @@
+"""Script for uploading the boot time measurement to influxdb"""
+
+import argparse
+import logging
+import mmap
+import re
+import sys
+import subprocess
+from datetime import datetime
+from influxdb import InfluxDBClient, exceptions
+import requests.exceptions
+
+def get_device():
+    """Get the device description for the current system (e.g. cRIO-90xx)"""
+    try:
+        process = subprocess.run(['fw_printenv', 'DeviceDesc'],
+                                 check=True,
+                                 stdout=subprocess.PIPE)
+        dev_str = process.stdout.strip().decode("utf-8")
+        dev = dev_str.split('=')[-1]
+    except FileNotFoundError:
+        logging.warning("failed to read the current device type")
+        raise
+
+    return dev
+
+
+def get_kernel_version():
+    """Get the kernel version currently running on the system"""
+    try:
+        process = subprocess.run(['uname', '-r'],
+                                 check=True,
+                                 stdout=subprocess.PIPE)
+        kver_full = process.stdout.strip().decode("utf-8")
+        rgx = re.search(r'([0-9]+\.[0-9]+)\.([0-9]+)', kver_full)
+        kver = rgx.group(1)
+    except (FileNotFoundError, AttributeError):
+        logging.warning("failed to read the current kernel version")
+        raise
+
+    return kver, kver_full
+
+
+def get_os_version():
+    """Extract OS version information from /etc/os-release"""
+    try:
+        with open('/etc/os-release', 'rb', 0) as file, \
+             mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as mfile:
+            rgx = re.search(br'VERSION_ID=([0-9.]+)', mfile)
+            version = rgx.group(1).decode()
+
+            rgx = re.search(br'BUILD_ID="([^"]+)"', mfile)
+            build = rgx.group(1).decode()
+
+            rgx = re.search(br'VERSION_CODENAME="([^"]+)"', mfile)
+            codename = rgx.group(1).decode()
+    except (ValueError, AttributeError):
+        logging.warning("failed to parse version information from '/etc/os-release'")
+        raise
+
+    return version, build, codename
+
+
+def get_boot_time(log_file):
+    """Extract boot time from log file"""
+    try:
+        with open(log_file, 'rb', 0) as file, \
+             mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ) as mfile:
+            rgx = re.search(br'([0-9.]+)', mfile)
+            boot_time = float(rgx.group(1).decode())
+    except (FileNotFoundError, ValueError, AttributeError):
+        logging.warning("failed to parse boot time from '%s'", log_file)
+        raise
+
+    return boot_time
+
+
+def read_data(path):
+    """Compile test data and metadata"""
+    controller = get_device()
+    kernel_version, kernel_full_version = get_kernel_version()
+    distro_version, distro_build, distro_name = get_os_version()
+    boot_time = get_boot_time(path)
+
+    print("boot time:", boot_time, "(s)")
+
+    data = {
+        "measurement": "boot_performance",
+        "tags": {
```

Comment:
Would it make sense to log a device hostname/serial number? I'm just imagining what kind of info would be useful if we see an unexpected change. Being able to trace a change to a recently swapped controller-under-test could be helpful.

Reply:
I don't expect large variations for the same controller model unless there's a hardware issue but I can see how having this info would make tracking a hardware issue easier so I'll add it.